### ISSUE TITLE
--threads always slows down the simulation

### ISSUE THREAD

**Issue by jotego** [verilator/verilator#2826]
When I try to use more threads by adding the `--threads` argument to the compilation; I always find that the simulation runs significantly slower. For instance:

--threads | Simulation Wall Time (s)
---------|-------------------
1  | 13 
2 |  38
4 | 1m43

I do not vary anything in my C++ test bench. I just add `--threads` and run the code. I think I must be missing something important as the threads just don't work as intended.

This is my Verilator version:
`Verilator 4.028 2020-02-06 rev v4.026-92-g890cecc1`

Could you help me, please?

-- Notes for me:
Command used on JT51:
`time go -f ghouls.vgz -time 2000`

**Comment by wsnyder** [comment]
Please try the most recent release as there's been many tracing changes and we prefer not to debug old releases, thanks.

**Comment by jotego** [comment]
Thanky you for your comment. I have compiled and installed the latest version:

`Verilator 4.199 devel rev v4.110-32-g9483ebefa`
I still see the same behaviour. Do you want me to give you the test case?

**Comment by wsnyder** [comment]
Please, ideally use test_regress format (see docs), but whatever you have, thanks.

**Comment by wsnyder** [comment]
Also please make sure you are using numactl as described in the docs.

**Comment by jotego** [comment]
Using `numactl -m 0 -C 0,1 -- ` in front of the simulation command rendered the same result. I will prepare the test case for you.

**Comment by wsnyder** [comment]
It probably does the same but also please try "numactl -m 0 -C 0,2".  On some systems 0 & 1 are SMT on same CPU.

**Comment by wsnyder** [comment]
Didn't see any additional updates so assuming you figured this out, feel free to post additional comments if needed.

**Comment by jotego** [comment]
Sorry for not answering back then, I tried and it didn't work and I just gave up.

A year later (today). I have tried again with a more complex design. Just in case the other one was simpler. I am simulating the whole Pang! arcade machine, including the FM chip, the CPU and the video system. I still find that the `--threads` option is significantly slower. The verilator version I'm using is "Verilator 4.224" and I have `numactl -m 0 -C 0,2` in front of the command.

No threads | threads = 2
-----|----
30s | 1m38

I have never seen Verilator perform faster during all the time I'm using it. It's a feature that really puzzles me.
